src/main/java/org/jenkinsci/plugins/gitclient/GitCredentialBindings.java

```diff
@@ -0,0 +1,20 @@
+package org.jenkinsci.plugins.gitclient;
+
+import com.cloudbees.plugins.credentials.common.StandardCredentials;
+import hudson.EnvVars;
+import hudson.FilePath;
+import hudson.model.TaskListener;
+
+import javax.annotation.Nonnull;
+import java.io.File;
+import java.io.IOException;
+
+public interface GitCredentialBindings {
+
+    void setKeyBindings(@Nonnull StandardCredentials credentials);
+
+    void setRunEnviornmentVariables(@Nonnull FilePath filePath, @Nonnull TaskListener listener) throws IOException, InterruptedException;
+
+    CliGitAPIImpl getCliGitAPIInstance(String gitExe, File workspace, TaskListener listener, EnvVars environment);
```

Comment:
@arpoch @MarkEWaite I am not sure if we should directly call the CliGitAPIImpl. The correct usage is to call [Git](https://github.com/jenkinsci/git-client-plugin/blob/master/src/main/java/org/jenkinsci/plugins/gitclient/Git.java) and then let it decide which implementation to use (in our case it will always be CliGitAPIImpl)

Reply:
with that change, the contract should have a `getGitClientInstance` or `getGitClient` which seems more suitable.

Reply:
Great suggestion @rishabhBudhouliya. Working on that currently will try to make a commit by today. Thanks.